src/widgets/table/table.rs

```diff
@@ -564,8 +566,39 @@ impl<'a> Table<'a> {
         reason = "The name for this feature is not final and may change in the future",
         issue = "https://github.com/ratatui-org/ratatui/issues/536"
     )]
-    pub const fn segment_size(mut self, segment_size: SegmentSize) -> Self {
-        self.segment_size = segment_size;
+    #[deprecated(note = "please use flex instead of segment_size")]
```

Comment:
We should add a `since = "v0.26.0"`.

```suggestion
#[deprecated(note = "please use Table::flex instead of Table::segment_size", since = "v0.26.0")]
```

Reply:
Yeah, you're right. It's definitely better

Reply:
Clippy didn't like something but I don't know what exactly

```
error: the since field must contain a semver-compliant version
   --> src/widgets/table/table.rs:570:9
    |
570 |         since = "v0.26.0",
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#deprecated_semver
    = note: `#[deny(clippy::deprecated_semver)]` on by default
```

Reply:
Sorry, remove the `v`.

Reply:
Yeah, now it works! Thanks! I updated that line to `#[deprecated(since = "0.26.0", note = "You should use Table::flex instead.")]`